lib/secure_log_formatter.ex

```diff
@@ -0,0 +1,91 @@
+import Kernel, except: [inspect: 1]
+
+defmodule SecureLogFormatter do
+  @moduledoc """
+  Documentation for SecureLogFormatter.
+  """
+
+  alias Logger.Formatter
+
+  @default_fields [~r/\w*_token/, "credit_card", "password"]
+  @default_pattern "\n$time $metadata[$level] $levelpad$message\n"
+
+  config = Application.get_env(:logger, :secure_log_formatter, [])
+  @fields Keyword.get(config, :fields, []) ++ @default_fields
+  @label Keyword.get(config, :replacement, "[REDACTED]")
+  @patterns Keyword.get(config, :patterns, [])
+  @format_config config
+    |> Keyword.get(:format, @default_pattern)
+    |> Formatter.compile
+
+  def default_fields, do: @default_fields
+
+  @doc """
+  Secure formatting for Elixir Logger
+
+  # Examples
+
+      iex> SecureLogFormatter.format(:info, "CC 4111111111111111", nil, [])
+      ["\\n", :time, " ", "", "[", "info", "] ", " ", "CC [REDACTED]", "\\n"]
+  """
+  def format(level, msg, ts, md),
+    do: Formatter.format(@format_config, level, sanitize(msg), ts, md)
+
+  @doc """
+  Securely inspect a value.
+
+  # Examples
+
+      iex> SecureLogFormatter.inspect(%{user: "username", password: "abc123"})
+      ~s(%{password: \"[REDACTED]\", user: \"username\"})
+
+      iex> SecureLogFormatter.inspect([access_token: "secret"])
+      ~s([access_token: \"[REDACTED]\"])
+
+      iex> SecureLogFormatter.inspect("Customer CC 4111111111111111")
+      ~s("Customer CC [REDACTED]")
+  """
+  def inspect(message) do
```

Comment:
@joshleecreates check the tests for how I handled this.  For now you need to do something like:


```elixir
import Kernel, except: [inspect: 1]
import SecureLogFormatter, only: [inspect: 1]
```